openedx/core/djangoapps/xblock/rest_api/views.py

```diff
@@ -211,6 +212,12 @@ def xblock_handler(
     block = load_block(usage_key, user, version=version)
     # Run the handler, and save any resulting XBlock field value changes:
     response_webob = block.handle(handler_name, request_webob, suffix)
+
+    if handler_name == "studio_submit":
+        # Signal that we've modified this block
+        context_impl = get_learning_context_impl(usage_key)
+        context_impl.send_block_updated_event(usage_key)
```

Comment:
Instead of putting this `send_block_updated_event()` here and [here](https://github.com/openedx/edx-platform/blob/master/openedx/core/djangoapps/xblock/rest_api/views.py#L325-L327), I think we should put it [here](https://github.com/openedx/edx-platform/blob/cd9b90fc213a535487f909a07c094b18d5466cac/openedx/core/djangoapps/xblock/runtime/learning_core_runtime.py#L315) in `save_block()`. Then it will be called consistently on _any_ handler that updates the fields, without assuming that only handlers called `studio_submit` are saving changes. And it will be called for changes via the REST API too.

Reply:
Yes! It's a good suggestion. I updated that here: https://github.com/openedx/edx-platform/pull/36029/commits/3b2268d0d9eec75960fc615ea9faeafbe2fc3a88